syno-plex-update.sh

```diff
@@ -10,13 +10,20 @@
 
 # ========== [Begin Configuration] ==========
 
+DSM_VERSION=$(cat /etc/VERSION | grep majorversion | tail -c 3 | head -c 1)
+OS_ARCHITECTURE="linux-$(uname -m)"
+
 # Path to the local Plex server preferences file (in NAS filesystem).
 # Can be found fia `sudo find / -name "Preferences.xml" | grep Plex`
 # Extracting and passing the online token seems optional though, so you can omit this.
-PLEX_PREFERENCES_FILE='/volume1/@apphome/PlexMediaServer/Plex Media Server/Preferences.xml'
+if [ "${DSM_VERSION}" -ge "7" ]; then
+    PLEX_PREFERENCES_FILE='/var/packages/PlexMediaServer/shares/PlexMediaServer/AppData/Plex Media Server/Preferences.xml'
+else
+    PLEX_PREFERENCES_FILE='/volume1/@apphome/PlexMediaServer/Plex Media Server/Preferences.xml'
+fi
 
 # Web endpoint for retrieving Plex release metadata.
-PLEX_RELEASE_API='https://plex.tv/api/downloads/5.json?X-Plex-Token=TokenPlaceholder'
+PLEX_RELEASE_API='https://plex.tv/api/downloads/5.json?channel=plexpass&X-Plex-Token=TokenPlaceholder'
```

Comment:
What is the benefit of using `channel=plexpass` here? I'd expect that not all users (including myself) use the paid services, so we should make sure free Plex users aren't locked out from using the package due to downloading an exclusive version. Alternatively, this could also be a config option to suit both types of users.

Reply:
I think for free users, X-Plex-Token is not required. Yes. it should be made into an option either by a config file or passed in as parameter.